### PR TITLE
feat(component): implements checkbox column type

### PR DESCRIPTION
**Pull request by jorgemoya**
## What
Implements checkbox column type

## Why
Ongoing development of Worksheet component.

## Beautiful stuff
Kapture 2021-04-26 at 11 31 07

### CODE REVIEW

File: packages/big-design/src/components/Worksheet/Cell/Cell.tsx [hunk 1]
```diff
@@ -89,13 +89,14 @@ const InternalCell = <T extends WorksheetItem>({
...
-      // Default to TextEditor
+      case 'checkbox':
+        return <CheckboxEditor cell={cell} onChange={handleChange} />;
       default:
         return isEditing ? (
           <TextEditor cell={cell} isEdited={isEdited} onBlur={handleBlur} onKeyDown={handleKeyDown} />
         ) : (
           // In case of NaN casting to string
-          <Small color="secondary70">{value.toString()}</Small>
+          <Small color="secondary70">{value ? value.toString() : ''}</Small>
```
Comment (jordan-massingill) [inline packages/big-design/src/components/Worksheet/Cell/Cell.tsx:99]: I think NaN will cast to "NaN" - is this `toString()` intended to avoid that?

Comment (jorgemoya) [reply packages/big-design/src/components/Worksheet/Cell/Cell.tsx:99]: Is it intended to do that.

Comment (jordan-massingill) [reply packages/big-design/src/components/Worksheet/Cell/Cell.tsx:99]: 👍 Ah, nvm, I was thinking of a different implementation (`||`) which would have returned `"NaN"`, but a ternary avoids that.

**Review by jordan-massingill — APPROVED** [review]
✅

File: packages/big-design/src/components/Worksheet/editors/CheckboxEditor/CheckboxEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  cell: Cell<Item>;
+  onChange(value: unknown): void;
+}
+
+const InternalCheckboxEditor = <T extends WorksheetItem>({ cell, onChange }: CheckboxEditorProps<T>) => {
+  const handleChange = () => {
+    onChange(!cell.value);
+  };
+
+  const checked = cell.value;
```
Comment (chanceaclark) [inline packages/big-design/src/components/Worksheet/editors/CheckboxEditor/CheckboxEditor.tsx:19]: Do we need this const declaration?

**Review by chanceaclark — COMMENTED** [review]
One small comment, but looks good.

File: packages/big-design/src/components/Worksheet/editors/CheckboxEditor/CheckboxEditor.tsx (hunk 2 above)
Comment (jorgemoya) [reply packages/big-design/src/components/Worksheet/editors/CheckboxEditor/CheckboxEditor.tsx:19]: I was thinking about removing it, so might as well.

**Review by chanceaclark — APPROVED** [review]

**Pull request merged**